src/id.rs

```diff
@@ -38,23 +39,79 @@ impl Id {
     #[doc(hidden)]
     #[track_caller]
     #[inline]
-    pub const unsafe fn from_u32(v: u32) -> Self {
+    pub const unsafe fn from_data(v: u32) -> Self {
```

Comment:
bikeshedding: The name doesn't make it clear that this is a new id without a generation. Maybe `new_without_generation` or `new_with_index`.